LSDR/Assets/Scripts/Audio/MusicSystem.cs

```diff
@@ -24,6 +25,14 @@ public AudioSource PlayRandomSongFromDirectory(string dir)
         public AudioSource PlayRandomSongFromDirectory(AudioSource source, string dir)
         {
             var clip = getRandomSongFromDirectory(dir);
+            
+            if (clip == source.clip)
```

Comment:
The indentation is off here, should be with 4 spaces for each level.
Additionally, I'm not sure if the while loop would ever exit as the AudioSource's clip doesn't change even when it finishes.
I think a better approach here would be to exclude the current song from the choices in getRandomSongFromDirectory(), this is easy with a LINQ expression.
For an example of this approach, please see the logic for selecting a new dream to link to (that isn't the current dream) in DreamSystem: https://github.com/Figglewatts/LSDRevamped/blob/master/LSDR/Assets/Scripts/Dream/DreamSystem.cs#L232-L234

Reply:
Can't figure out how to change this into a linq expression, sorry.
The while loop does exit just fine, I test my code before submitting. The while loop only fails to exit if there's only one song in the directory.